src/main/java/dev/waterdog/waterdogpe/network/downstream/ConnectedDownstreamHandler.java

```diff
@@ -45,6 +46,22 @@ public final boolean handle(RemoveObjectivePacket packet) {
         return false;
     }
 
+    @Override
+    public final boolean handle(SetScorePacket packet) {
+        for(ScoreInfo info : packet.getInfos()) {
+            this.player.getScoreInfos().stream().
+                    filter(info1 -> info1.getScoreboardId() == info.getScoreboardId()).
+                    findFirst().ifPresent(info1 -> this.player.getScoreInfos().remove(info1));
+        }
```

Comment:
Removing score infos with same scoreboard id

Reply:
Can you replace the stream with a normal loop? Streams are quite slow with small data sets from my experience

Reply:
Fixed and I knew that but was a little too lazy.